publication/src/main/scala/hmda/publication/reports/util/ReportsMetadataLookup.scala

```diff
@@ -0,0 +1,49 @@
+package hmda.publication.reports.util
+
+import com.github.tototoshi.csv.CSVParser.parse
+import hmda.model.ResourceUtils
+import hmda.model.publication.reports.ReportTypeEnum
+import hmda.model.publication.reports.ReportTypeEnum.Disclosure
+import hmda.publication.reports.util.DispositionType.{ DispositionType, ReceivedDisp }
+
+object ReportsMetaDataLookup extends ResourceUtils {
+
+  private val lines = resourceLines("/reports-metadata.txt")
+
+  val values: Map[String, ReportMetaData] = lines.drop(1).map { line =>
+    val data = ReportMetaData.fromString(line)
+    data.reportId -> data
+  }.toMap
+
+}
+
+case class ReportMetaData(
+  reportId: String,
+  reportType: ReportTypeEnum,
+  reportTable: String,
+  dispositions: List[DispositionType],
+  description: String
+)
+
+case object ReportMetaData {
+
+  def fromString(line: String): ReportMetaData = {
+    val values = parse(line, '\\', ',', '"').getOrElse(List())
+    val reportId = values.head
+    val reportType = ReportTypeEnum.byName.getOrElse(values(1).toLowerCase, Disclosure)
```

Comment:
I always prefer using `getOrElse()`, since it's always safe... But I'm not sure if it's right in this case.  If we create the metadata file incorrectly, it might actually be more useful for this line to fail on a plain `get()`.  Same on line 36.

Reply:
I wondered about that too. That works for me! Updated in c64cc9a544786bb21cbf3a3de3e8d4be78ed42fa.